scripts/output/export/xlsx_IIASA.R

```diff
@@ -9,117 +9,134 @@ require(data.table)
 require(iamc)
 require(rmndt)
 library(dplyr)
-library(tidyr)
 library(quitte)
-library(readxl)
-library(yaml)
-library(magclass)
 library(lucode2)
+library(magclass)
 library(magpie4)
-library(stringr) # str_sub
+library(piamInterfaces)
+library(stringr) # str_sub, str_split
+library(tibble)
+library(tidyr)
+
+meslog <- function(text, LOGFILE) {
+  message(text)
+  write(text, file = LOGFILE, append = TRUE)
```

Comment:
There's also `sink` which can achieve a similar thing: Run `sink(LOGFILE, append = TRUE, type = "message", split = TRUE)`, then all calls to `message` will write to terminal (actually `stderr`) and the logfile. You must remember to call `sink()` at the end of the script/when you want to stop redirecting output to the logfile.

Reply:
`split = TRUE` only splits R output (via Rvprintf) and the default output from writeLines: it does not split all output that might be sent to stdout. (https://stat.ethz.ch/R-manual/R-devel/library/base/html/sink.html)

So I don't think it works here…

Reply:
Could be, although in my experience sink usually works as expected. I'm a little afraid that we will see `meslog` or similar functions defined over and over in different scripts, in that case I'd much prefer pulling meslog up into e.g. the gms package.

Reply:
please try `sink(file("test.log", "w"), type = "message", split = TRUE)` (without the `file` part, it also crashes):
```
Error in `sink()`:
! cannot split the message connection
```
Happy to add `meslog` or something similar to `gms`.

Reply:
Okay I'm convinced `sink` - does not like messages. I just remembered another way:
```r
withCallingHandlers({
  # all the code that should message to console and log
}, message = function(x) {
  cat(x$message, file = LOGFILE, append = TRUE)
  message(x$message) # not sure, maybe need to use cat here
})
```
This is better than the sink approach anyway, because you cannot forget to run `sink()` at the end. This is also similar to what [madrat does](https://github.com/pik-piam/madrat/blob/master/R/withMadratLogging.R).